wgpu-hal/src/gles/egl.rs

```diff
@@ -1052,6 +1082,8 @@ impl crate::Instance for Instance {
             unsafe { gl.debug_message_callback(super::gl_debug_message_callback) };
         }
 
+        // Avoid accidental drop when the context is not current.
+        let gl = ManuallyDrop::new(gl);
```

Comment:
Is this a stale comment? You're not avoiding a drop when the context is _not current_, instead you're _forcing it to be current_ Before dropping.

Reply:
This comment is noting the reason to wrap in `ManuallyDrop` before calling `unmake_current` on the line below. This prevents the `glow::Context` from being dropped if there is some panic between when we call `unmake_current` and when `gl` is placed into the `AdapterContext` (where the drop impl will handle making the context current before dropping).

FWIW the comment could probably be clarified.

Reply:
Clarification is happening in https://github.com/gfx-rs/wgpu/pull/6152 where I ran into a few conflicts with this PR. Panics (except inside the drop handler) [should still cause `Drop` to be called while unwinding](https://doc.rust-lang.org/nomicon/unwinding.html). I'll add a mention that the only skipped case is between `let gl` and when `Self {}` is finally constructed.

Fwiw when a debug callback is specified by the user (in this case `wgpu`), it's common to skip messages when panicking:

https://github.com/gfx-rs/wgpu/blob/5deaef3b67259008844b15e1fd1d3a8924d1cde0/wgpu-hal/src/vulkan/instance.rs#L21-L23

Not sure if that can be done in EGL/WGL as well?

Reply:
> Panics (except inside the drop handler) [should still cause Drop to be called while unwinding](https://doc.rust-lang.org/nomicon/unwinding.html)

Yes, this is why it needs to be wrapped in `ManuallyDrop` in the intermittent time before `AdapterContext` is constructed since panics would drop the `glow::Context` which could lead to UB if it isn't current.

> Fwiw when a debug callback is specified by the user (in this case wgpu), it's common to skip messages when panicking:

Something like this won't help here with the way the `Drop` impl for `glow::Context` works, by the time the callback is invoked there is already UB.